provisioner/aws-manager.js

```diff
@@ -70,15 +102,75 @@ AwsManager.prototype.update = function () {
         },
       ],
     }),
+    that.ec2.describeInstances({
+      Filters: [
+        {
+          Name: 'key-name',
+          Values: [that.keyPrefix + '*'],
+        },
+        {
+          Name: 'instance-state-name',
+          Values: ['shutting-down', 'terminated', 'stopping'],
+        },
+      ],
+    }),
+    that.ec2.describeSpotInstanceRequests({
+      Filters: [
+        {
+          Name: 'launch.key-name',
+          Values: [that.keyPrefix + '*'],
+        }, {
+          Name: 'state',
+          Values: ['cancelled', 'failed', 'closed', 'active'],
+        },
+      ],
+    }),
   ]);
 
   p = p.then(function (res) {
-    if (res[0].NextToken && res[0].NextToken !== '') {
+    var livingInstances = res[0];
+    var livingSpotRequests = res[1];
+    var deadInstances = res[2];
+    var deadSpotRequests = {};
+
+    // NOTE: Dead spot request does not mean that the spot request failed,
+    //       just that it's no longer an open spot request...
+
+    // TODO: Probably better to do a single set of describe calls and bucket
+    //       them in this section... hmm, no strong feelings until data
+
+    // Because we don't filter the spot requests, we'll need to make sure that
+    // we remove the SpotInstanceRequests per-region response so that
+    // that._classify() knows how to interpret the data
+    Object.keys(res[3]).forEach(function (region) {
+      deadSpotRequests[region] = res[3][region].SpotInstanceRequests;
+    });
+
+    // We don't currently handle this case because it seems to be deprecated in
+    // the docs.  We should be alerted if it begins to be used again
+    if (livingInstances.NextToken && livingInstances.NextToken !== '') {
       console.log('[alert-operator] WARNING!  We have a ' +
           'DescribeInstances NextToken and arent using it!!!');
-    };
-    var filteredSpotRequests = that._filterSpotRequests(res[1]);
-    that.__apiState = that._classify(res[0], filteredSpotRequests.good);
+    }
+    var filteredSpotRequests = that._filterSpotRequests(livingSpotRequests);
+
+    // We store the last iteration's state
+    that.__previousApiState = that.__apiState;
+    that.__apiState = that._classify(livingInstances, filteredSpotRequests.good);
+    that.__deadState = that._classify(deadInstances, deadSpotRequests);
+
+    // remove this try/catch once this has been live for a while
+    try {
+      var stateDifferences = that._compareStates(that.__apiState, that.__previousApiState, that.__deadState);
+      that._reconcileStateDifferences(stateDifferences, that.__deadState, that.__apiState);
+    } catch(err) {
+      console.log('[alert-operator] failure in computing differences');
+      console.log(err);
+      if (err.stack) {
+        console.log(err.stack);
+      }
+    }
+
     return that.handleStalledRequests(filteredSpotRequests.stalled);
   });
 
```

Comment:
Don't know exactly where to put this comment. But we should use information about dead spot requests in `_reconcileInternalState`, so that it's even less likely that we have to fallback and rely on the spot request timeout thing...